Enable Audio feature.

Sunwoo-Nam opened the pull request:
1. Connect with TCP socket instead of SyncSocket.
2. Sync a shared memory for audio data from RP to BP.

Sunwoo-Nam commented:
I'm refactoring this PR. Please do not review until update.

Sunwoo-Nam commented:
Now, the renderer process creates a TCP server and propagates the ip address and port number to the browser process for TCP connection.

younghajung commented on mojo/public/cpp/platform/tcp_platform_handle_utils_posix.cc:
```diff
@@ -168,4 +169,23 @@ bool TCPServerAcceptConnection(const base::PlatformFile server_socket,
+  struct hostent* host_entry;
+  char hostname[80];
+  gethostname(hostname, sizeof(hostname));
+  host_entry = gethostbyname(hostname);
```
In my local machine, `gethostbyname` just returns 127.0.1.1.
I think `gethostbyname` is based on hosts file.
Hence, if there is no information about own IP address in hosts file, we cannot get own address.
How about using `getifaddrs`?

Sunwoo-Nam replied:
I think It's better to remove this code :)
I am going to update PR. Thank you.

Sunwoo-Nam committed:
Enable Audio feature.

1. Connect with TCP socket instead of SyncSocket.
2. Sync a shared memory for audio data from RP to BP.

Signed-off-by: Sunwoo Nam <sunny.nam@samsung.com>

is46-kim commented on content/renderer/media/audio/mojo_audio_output_ipc.cc:
```diff
@@ -233,6 +237,17 @@ void MojoAudioOutputIPC::Created(
+#if defined(CASTANETS)
+  // If socket_handle is invalid.
+  if (socket_handle < 0) {
+    shared_memory_region_ = std::move(shared_memory_region);
```
How about passing |shared_memory_region| with parameter to callback function without |shared_memory_region_| member?

Sunwoo-Nam replied:
Thank you for your point.

hyowon reviewed (approved):
It would be better if the audio service does not directly control the connection of SyncSocket. But let's merge this PR as audio works well with this implementation.
